echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EventContent.java

```diff
@@ -44,25 +53,51 @@ public EventContent(Event event, String type) throws FieldNotFoundException {
   }
 
   private void setRepo() throws FieldNotFoundException {
-    repo =
+    Map trigger =
         Optional.ofNullable(event.getContent())
             .map(content -> (Map) content.get("execution"))
             .map(execution -> (Map) execution.get("trigger"))
-            .map(trigger -> (Map) trigger.get("buildInfo"))
-            .map(buildInfo -> (String) buildInfo.get("name"))
             .orElseThrow(FieldNotFoundException::new);
+    if (trigger.containsKey("type")
+        && trigger.get("type").equals("git")
+        && trigger.get("source").equals("github")) {
+      String repo_project =
+          Optional.ofNullable((String) trigger.get("project"))
+              .orElseThrow(FieldNotFoundException::new);
```

Comment:
As someone who occasionally has to stare at echo logs, I would love a more descriptive error message here, for instance:

          Optional.ofNullable((String) trigger.get("project"))
              .orElseThrow(() -> new FieldNotFoundException("Could not find field `project` in event"));

And as I'm saying this, I realize that `FieldNotFoundException` does not have such a constructor, but it would be great if someone were to add it for the greater good ❤️